pjlib/include/pj/atomic_queue.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 Teluu Inc. (http://www.teluu.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJ_ATOMIC_QUEUE_H__
+#define __PJ_ATOMIC_QUEUE_H__
+
+/**
+ * @file atomic_queue.h
+ * @brief Atomic Queue operations
+ */
+
+#include <pj/types.h>
+
+PJ_BEGIN_DECL
+
+/* Atomic queue (ring buffer) for single consumer & single producer.
+ *
+ * Producer invokes 'pj_atomic_queue_put(item)' to put an item to the back of
+ * the queue.
+ * Consumer invokes 'pj_atomic_queue_get(item)' to get an item from the head of
+ * the queue.
+ */
```

Comment:
This does not seem to be part of the docs?
Note that the "single consumer & single producer" condition needs to be explicitly specified in the docs.